src/libcfitsio.jl

```diff
@@ -120,8 +120,12 @@ export FITSFile,
 using Libdl
 
 import Base: FastContiguousSubArray
-const ArrayOrFastContiguousSubArray{T} = 
-      Union{Array{T},FastContiguousSubArray{T,N,<:Array{T}} where N}
+
+const ArrayOrFastContiguousSubArray{T,N} = Union{Array{T,N},
+      FastContiguousSubArray{T,N,<:Array{T}}}
+
+const ContiguousAbstractArray{T,N} = Union{ArrayOrFastContiguousSubArray{T,N},
+      Base.ReinterpretArray{T,N,Complex{T},<:ArrayOrFastContiguousSubArray{Complex{T},N}}}
```

Comment:
I think we basically want to use [`StridedArray`](https://docs.julialang.org/en/v1/base/arrays/#Base.StridedArray)?

Reply:
This is the definition: https://github.com/JuliaLang/julia/blob/381693d3dfc9b7072707f6d544f82f6637fc5e7c/base/reinterpretarray.jl#L48-L54.  The only thing that bugs me about this PR is that we keep adding unions over unions, it feels like we should be able to do something better.

Reply:
I agree about the unions, even I had wondered about this. Open for suggestions

Reply:
@kbarbary any opinion on using `StridedArray` everywhere?

Reply:
@jishnub I'm more and more convinced that using `StridedArray` is what we want.  My understanding is that it describes the kind of arrays we want to pass to the library, as also suggested by its docstring.  Would you mind updating the PR to use `StridedArray`?

Reply:
I've tried to implement this, but non-contiguous arrays don't seem to work even though they might be a `StridedArray`. Is there a way to ensure that a `StridedArray` is contiguous? Or otherwise is there a way to pass the stride to cfitsio? I'm not sure if the latter is possible at first glance, as many of the cfitsio functions seem to read rectangular sections from an image.

At the moment I end up with incorrect results with non-contiguous views such as 

```julia
julia> fitsfile=tempname();

julia> indata = reshape(Float32[1:6;], 2, 3)
2×3 Array{Float32,2}:
 1.0  3.0  5.0
 2.0  4.0  6.0

julia> FITS(fitsfile,"w") do f
       write(f,indata)
       end;

julia> b = zero(indata);

julia> b_view = view(b,1,:); # non-contiguous

# Expected output
julia> FITS(fitsfile,"r") do f
       read(f[1],1,:)
       end
3-element Array{Float32,1}:
 1.0
 3.0
 5.0

# Actual output
julia> FITS(fitsfile,"r") do f
       read!(f[1],b_view,1,:)
       end
3-element view(::Array{Float32,2}, 1, :) with eltype Float32:
 1.0
 5.0
 0.0

# Make sure that we're dealing with the correct types
julia> b_view isa StridedArray
true

julia> b_view isa Base.StridedSubArray
true
```

Perhaps I'm missing something here as I'm new to `StridedArray`s, so I'm open to suggestions

Reply:
Ok, nevermind, thanks for testing!